tests/app/main/helpers/test_frameworks.py

```diff
@@ -364,3 +369,148 @@ def test_check_agreement_is_related_to_supplier_framework_or_abort_does_not_abor
     supplier_framework = {"supplierId": 212, "frameworkSlug": 'g-cloud-8'}
     agreement = {"supplierId": 212, "frameworkSlug": 'g-cloud-8'}
     check_agreement_is_related_to_supplier_framework_or_abort(agreement, supplier_framework)
+
+
+def test_order_frameworks_for_reuse():
+    """Test happy path. Should return 2 frameworks, closest date first."""
+    t09 = '2009-03-03T01:01:01.000000Z'
+    t07 = '2007-03-03T01:01:01.000000Z'
+    t12 = '2012-03-03T01:01:01.000000Z'
+
+    fake_frameworks = [
+        {'allow_declaration_reuse': True, 'application_close_date': t12, 'extraneous_field': 'foo'},
+        {'allow_declaration_reuse': False, 'application_close_date': t09, 'extraneous_field': 'foo'},
+        {'allow_declaration_reuse': True, 'application_close_date': t07, 'extraneous_field': 'foo'},
+    ]
+    ordered = order_frameworks_for_reuse(fake_frameworks)
+
+    assert ordered == [
+        {'allow_declaration_reuse': True, 'application_close_date': t12, 'extraneous_field': 'foo'},
+        {'allow_declaration_reuse': True, 'application_close_date': t07, 'extraneous_field': 'foo'}
+    ]
+
+
+def test_order_frameworks_for_reuse_filters():
+    """Test that this filters out anything with `allow_declaration_reuse == False`."""
+    t09 = '2009-03-03T01:01:01.000000Z'
+    t07 = '2007-03-03T01:01:01.000000Z'
+    t12 = '2012-03-03T01:01:01.000000Z'
+
+    fake_frameworks = [
+        {'allow_declaration_reuse': True, 'application_close_date': t12, 'extraneous_field': 'foo'},
+        {'allow_declaration_reuse': False, 'application_close_date': t09, 'extraneous_field': 'foo'},
+        {'allow_declaration_reuse': True, 'application_close_date': t07, 'extraneous_field': 'foo'},
+    ]
+    ordered = order_frameworks_for_reuse(fake_frameworks)
+
+    assert len(ordered) == 2
```

Comment:
It feels like the first test already covers this, and does more because it also checks that the "right" two frameworks are returned.